src/Pipe.re

```diff
@@ -0,0 +1,10 @@
+type t('a) = ref(option('a));
```

Comment:
`Pipe` might be a bit of an overloaded name:
1) It sounds like it could be associated with `Stream`
or
2) It sounds like it could be related to a unix named pipe (like from `Unix.pipe`).

1) will go away once we get rid of Stream though. I don't have a better suggestion though so we can leave as-is - just thinking out loud. I think your description goes a long way in explaining it's purpose as it satnds.

And it's not even part of our public API, so less of a concern...

Reply:
Definitely. I considered a few other names, like "channel" and "port", but "pipe" is the most accurate name I could think of, as it describes a single object with two ends that you can pass things through. While channel seems even more like a stream, which things "flow" through conitnuously, and port is more like just one end of a pipe.

There's also "tube", which apparently is stronger and more expensive than pipes I've learned, but I still prefer "pipe" for vague reasons that elude me. Either term could also be qualified to distinguish them, like `TypedTube` for example, except that's still way too generic.

I'm sure we'll eventually find a better name for it if we need it elsewhere. As you say, there's no rush.